src/test/java/com/hazelcast/aws/AwsDiscoveryStrategyFactoryTest.java

```diff
@@ -90,7 +90,7 @@ public void testFull() throws Exception {
         props.put("tag-key", "test-value");
         props.put("tag-value", "test-value");
         props.put("connection-timeout-seconds", 10);
-        props.put("hz-port", 1234);
+        props.put("hz-port", "1234");
```

Comment:
just wondering if this breaks backward compatibility (for code like this, where someone used an integer for a single port and now a string is required)... maybe worth adding a test to clarify?

Reply:
That is a super valid comment. Actually, it was not backwards compatible! I've changed it (which made the code slightly worse, but keeps the backward compatibility. Thanks a lot for this comment :)